tracing-subscriber/src/filter/env/mod.rs

```diff
@@ -193,54 +226,175 @@ impl EnvFilter {
     ///
     pub const DEFAULT_ENV: &'static str = "RUST_LOG";
 
+    /// Returns a [builder] that can be used to configure a new [`EnvFilter`]
+    /// instance.
+    ///
+    /// The [`Builder`] type is used to set additional configurations, such as
+    /// [whether regular expressions are enabled](Builder::with_regex) or [the
+    /// default directive](Builder::with_default_directive) before parsing an
+    /// [`EnvFilter`] from a string or environment variable.
+    ///
+    /// [builder]: https://rust-unofficial.github.io/patterns/patterns/creational/builder.html
+    pub fn builder() -> Builder {
+        Builder::default()
+    }
+
     /// Returns a new `EnvFilter` from the value of the `RUST_LOG` environment
     /// variable, ignoring any invalid filter directives.
+    ///
+    /// If the environment variable is empty or not set, or if it contains only
+    /// invalid directives, a default directive enabling the [`ERROR`] level is
```

Comment:
does this _need_ to be `ERROR`, or would a switch to, e.g., `INFO` as the default be an unacceptable consistency break with other functions that create a layer/subscriber?

EDITL yeah, it can't happen now.

Reply:
it has to be `ERROR` for compatibility. IMO, the best default is probably *nothing*, now that it's settable --- when you create a new builder, it defaults to not having a default directive, but for the constructors we need to retain previous behavior.